core/src/navigation/services/semi-collapsed-navigation.js

```diff
@@ -7,11 +7,21 @@ class SemiCollapsibleNavigationClass {
     this.responsiveNavSetting = LuigiConfig.getConfigValue(
       'settings.responsiveNavigation'
     );
-    this.semiCollapsible =
-      this.responsiveNavSetting === 'semiCollapsible' ||
-      this.responsiveNavSetting === 'Fiori3'
-        ? true
-        : false;
+    let currentCollapsedState = localStorage.getItem(
+      NavigationHelpers.COL_NAV_KEY
+    );
+    //checking if here was a previous state in localStorage before the reload
+    if ((currentCollapsedState = true)) {
```

Comment:
Probably you want to check if `currentCollapsedState === true` but than it will not work anymore.
My suggestion is to check if `currentCollapsedState` is not `null`. If it is **not** null you go with the state from the localstorage, if it is null you do the else part.

Reply:
updated the PR